### PR TITLE
Add "enrolled" field to requests collection definition

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR adds the "enrolled" field to the `requests` collection definition.

## 💭 Motivation and context ##

Related to the changes in https://github.com/cisagov/cyhy-core/pull/86 for https://github.com/cisagov/cyhy-core/issues/85.
 

## 🧪 Testing ##

Tested via visual inspection.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: NCATS_Data_Dictionary.md [hunk 1]
```diff
@@ -385,6 +385,8 @@ stakeholders.
     "TRIBAL", "TERRITORIAL", "PRIVATE")
 - `children` [list of strings]: Identifiers of organizations that are
   children of this organization
+- `enrolled` [ISO date]: Timestamp when this organization was enrolled
```
Comment (mcdonnnj) [inline NCATS_Data_Dictionary.md]: I feel like it should be this but I see the document has similar field descriptions with and without the "of".

```suggestion
- `enrolled` [ISO date]: Timestamp of when this organization was enrolled
```

**Review by mcdonnnj — APPROVED** [review]
Thank you for updating the docs! One item for your consideration.

File: NCATS_Data_Dictionary.md (hunk 1 above)
Comment (dav3r) [reply NCATS_Data_Dictionary.md]: I think either way is acceptable.  Just going to leave it as is, if that's alright with you.

Comment (jsf9k) [reply NCATS_Data_Dictionary.md]: I think `Timestamp indicating when this organization was enrolled` is superior to the way it is now and the suggestion, but I don't care enough to cause a ruckus.

**Review by jsf9k — APPROVED** [review]

File: NCATS_Data_Dictionary.md (hunk 1 above)
Comment (dav3r) [reply NCATS_Data_Dictionary.md]: Alright, you sold me on it.  Please see https://github.com/cisagov/ncats-data-dictionary/pull/44/commits/f0165745266aa2ffd125503fe2d8267323e78de8.

Comment (jsf9k) [reply NCATS_Data_Dictionary.md]: ~sold me~ strong-armed me :laughing:

Comment (dav3r) [reply NCATS_Data_Dictionary.md]: @mcdonnnj Is this conversation good to be resolved?

Comment (mcdonnnj) [reply NCATS_Data_Dictionary.md]: Yep. Sorry missed the update.

**Pull request merged**